### PR TITLE
e2e - Improve namespace deletion 

### LINKED ISSUE #2646: [FLAKE] can satisfy an associated ClusterServiceVersion's ownership requirement
### Description

Unfortunately, I marked this as a flake without capturing the stack trace ._.
Once I get it I'll update the issue. Or if this test fails for you, please paste in the stack trace and log link if possible.

### PR DESCRIPTION
**Pull request by akihikokuroda**
Cleanup before the namespace deletion and increase the delete wait timeout in some places.

Signed-off-by: akihikokuroda <akihikokuroda2020@gmail.com>

**Description of the change:**
Removing some remaining resources before the namespace deletion
Use dynamic client to make sure the resources are deleted in TearDown method
Increase some namespace deletion timeouts that often fails in e2e tests. 

**Motivation for the change:**

May fix #2646, #2641, and #2639
**Reviewer Checklist**
- [ ] Implementation matches the proposed design, or proposal is updated to match implementation
- [ ] Sufficient unit test coverage
- [ ] Sufficient end-to-end test coverage
- [ ] Docs updated or added to `/doc`
- [ ] Commit messages sensible and descriptive
- [ ] Tests marked as `[FLAKE]` are truly flaky
- [ ] Tests that remove the `[FLAKE]` tag are no longer flaky

### CODE REVIEW

File: test/e2e/csv_e2e_test.go [hunk 1]
```diff
@@ -4249,11 +4258,19 @@ var _ = Describe("Disabling copied CSVs", func() {
...
 			return nil
 		}).Should(Succeed())
+		Eventually(func() error {
+			err := ctx.Ctx().Client().Delete(context.Background(), &og)
+			if err != nil && k8serrors.IsNotFound(err) {
+				return err
+			}
+
+			return nil
+		}).Should(Succeed())
```
Comment (awgreene) [inline test/e2e/csv_e2e_test.go:4268]: I was unaware that failing to delete the OperatorGroup within a namespace prior to deleting the namespace could cause timeout issues. This is unintuitive to me given that OperatorGroups are a namespaced resource that shouldn't require much overhead to delete, do we know the cause?

File: test/e2e/util.go [hunk 2]
```diff
@@ -420,29 +420,29 @@ func TearDown(namespace string) {
...
 	}).Should(Succeed(), "failed to delete test installplans")
 
-	Eventually(func() (remaining []operatorsv1alpha1.InstallPlan, err error) {
-		list := &operatorsv1alpha1.InstallPlanList{}
-		err = client.List(clientCtx, list, inNamespace)
-		if list != nil {
-			remaining = list.Items
+	var installPlangvr = schema.GroupVersionResource{Group: "operators.coreos.com", Version: "v1alpha1", Resource: "installplans"}
+	Eventually(func() ([]unstructured.Unstructured, error) {
+	fmt.Printf("checking installplan\n")
```
Comment (awgreene) [inline test/e2e/util.go:425]: Is there a reason you chose to use the `fmt` package here?
```suggestion
	logf("checking installplan\n")
```

File: test/e2e/util.go [hunk 3]
```diff
@@ -420,29 +420,29 @@ func TearDown(namespace string) {
...
 	}).Should(Succeed(), "failed to delete test catalogsources")
 
-	Eventually(func() (remaining []operatorsv1alpha1.CatalogSource, err error) {
-		list := &operatorsv1alpha1.CatalogSourceList{}
-		err = client.List(clientCtx, list, inNamespace, ephemeralCatalogFieldSelector)
-		if list != nil {
-			remaining = list.Items
+	var catalogSourcegvr = schema.GroupVersionResource{Group: "operators.coreos.com", Version: "v1alpha1", Resource: "catalogsources"}
+	Eventually(func() ([]unstructured.Unstructured, error) {
+	fmt.Printf("checking catalogsource\n")
```
Comment (awgreene) [inline test/e2e/util.go:440]: Using `fmt` here as well, can be closed / changed based on your reasoning [here](https://github.com/operator-framework/operator-lifecycle-manager/pull/2676/files#r821794321).

File: test/e2e/util.go (hunk 2 above)
Comment (akihikokuroda) [reply test/e2e/util.go:425]: Sorry, it was for my debug.  I'll take out.

File: test/e2e/util.go (hunk 3 above)
Comment (akihikokuroda) [reply test/e2e/util.go:440]: Sorry, it was for my debug.  I'll take out.

File: test/e2e/csv_e2e_test.go (hunk 1 above)
Comment (akihikokuroda) [reply test/e2e/csv_e2e_test.go:4268]: Actually I don't know if the OperatorGroup is causing the error.  I delete it to eliminate all possible cause of the namespace deletion hung. It is the only resource remaining in the namespace except default service accounts and secrets that all namespace have.

File: test/e2e/csv_e2e_test.go [hunk 4]
```diff
@@ -4249,11 +4258,19 @@ var _ = Describe("Disabling copied CSVs", func() {
 
 			return nil
 		}).Should(Succeed())
+		Eventually(func() error {
+			err := ctx.Ctx().Client().Delete(context.Background(), &og)
+			if err != nil && k8serrors.IsNotFound(err) {
+				return err
+			}
```
Comment (perdasilva) [inline test/e2e/csv_e2e_test.go:4265]:
```suggestion
			if err := client.IgnoreNotFound(ctx.Ctx().Client().Delete(context.Background(), &og)); err != nil {
				return err
			}
```

Comment (perdasilva) [reply test/e2e/csv_e2e_test.go:4265]: Something I learned recently ^^

File: test/e2e/csv_e2e_test.go (hunk 1 above)
Comment (perdasilva) [reply test/e2e/csv_e2e_test.go:4268]: Hopefully when we integrate the new client, resource clean-up will be ezpz!

**Comment by perdasilva** [comment]
closing due to staleness - @akihikokuroda if you still want to get this change down, could you please resolve conflicts and re-open?

**Pull request closed without merging**